### PR TITLE
Much more capable and complete operator[].

### PR DESCRIPTION
**Pull request by staleyLANL**
This should help immensely with the high-level interface.
Full tests for operator[].
That was tons of careful work. I need a vacation.
Fortunately, it's Christmas Eve right now.

### CODE REVIEW

**Comment by staleyLANL** [comment]
@jlconlin @whaeck I'll probably have more material upload in a day or two, so feel free to take a look at this whenever you get the chance.

**Comment by staleyLANL** [comment]
@jlconlin @whaeck If and when this is good to go, I have more material I can upload.

File: src/GNDStk/Node/test/subscript.test.cpp [hunk 1]
```diff
@@ -1,26 +1,691 @@
...
+            // Actually writes into the tree...
+            EvaluatedNode[myversion] = "123.456.789";
+            CHECK(tree(reactionSuite,styles,evaluated,basic::version) ==
+                  "123.456.789");
+            // can still use as rvalue, and implicitly converts to string...
+            std::string foo = EvaluatedNode[myversion];
+            CHECK(foo == "123.456.789");
+
+            // Also actually writes into the tree...
+            EvaluatedNode[myversion] = version_t(12,34,56);
```
Comment (whaeck) [inline src/GNDStk/Node/test/subscript.test.cpp:289]: I assume the prerequisite for this to work is to have an ostream operator>> defined for version_t?

Builtin types like double, int, etc. have this by default. We should discuss precision in floating points at some point.

**Review by whaeck — APPROVED** [review]
Some renaming arguments, assigning default arguments. The main work here is operator[], tests were added for this to illustrate the functionality. Good to go for me.

File: src/GNDStk/Node/test/subscript.test.cpp (hunk 1 above)
Comment (staleyLANL) [reply src/GNDStk/Node/test/subscript.test.cpp:289]: Re: version_t, the prerequisite is that it have the usual GNDStk `convert()` business. However, my built-in `convert()` functions for `meta_t` eventually fall back on stream I/O if all else fails.

Comment (staleyLANL) [reply src/GNDStk/Node/test/subscript.test.cpp:289]: Let's come up with a short example in which we write a couple of small sections of GNDStk data, and want to set precision individually in each one.

**Pull request merged**